lib/msal-react/src/useIsAuthenticated.ts

```diff
@@ -1,21 +1,36 @@
-import { useState, useEffect } from 'react';
+import { AccountInfo, IPublicClientApplication } from "@azure/msal-browser";
+import { useState, useEffect } from "react";
 
-import { useMsal } from './MsalProvider';
-import { isAuthenticated } from './utilities';
+import { useMsal } from "./MsalProvider";
 
-export function useIsAuthenticated(username?: string): boolean {
+function isAuthenticated(instance: IPublicClientApplication, account?: Partial<AccountInfo>): boolean {
```

Comment:
This probably belongs in msal-browser. Open questions: 
- Should we create a new type containing optional parameters: username, homeAccountId, and possibly accountId in the future. Or what's shown here an optional AccountInfo object?
- Should the lookup logic try to match every parameter provided, or give precedence to certain params as shown here?